Feat: adds app.bsky.graph.defs#skeletonReasonSponsored lexicon.

aendra-rininsland opened the pull request:
See #3386.

I'm also working on a PR for bluesky-social/social-app to add UI for these. Creating this as a draft in the interim.

aendra-rininsland commented:
Marking as "ready" given this needs to be merged before bluesky-social/social-app#7518.

bnewbold commented on lexicons/app/bsky/feed/defs.json:
```diff
@@ -205,6 +209,15 @@
+      "properties": {
+        "info": {
+          "type": "string",
+          "description": "Extended information about sponsored content, provided by the feed generator for transparency purposes."
```
Can probably hold off on changes until the `social-app` PR gets a reply, but quick notes:

- a description on the overall object clarifying that it existing at all means the feed item is sponsored content (we don't have good descriptions like this for `skeletonReasonRepost` or `skeletonReasonPin`, but we should)
- we are slowly but surely trying to lock down every `string` field with a `maxLength` to prevent shenanigans. maybe 2000 bytes (like `feedContext`)?

I'd be interested in considering if this should eventually be an AT-URI field pointing at a "context" record of some type. We are thinking about a generic "annotation" / "more context" mechanism for things like labels. That is still speculative though, and this is an optional field so there is a transition path.

aendra-rininsland replied:
Thanks for the feedback! I've added a description and a maxLength as requested in the meantime while waiting for a response on bluesky-social/social-app#7518.

aendra-rininsland commented:
I've added two more fields, promotionCreator and promotionManager, to show who is running and benefitting from a promotional campaign. I've also renamed it from `skeletonReasonSponsored` to `skeletonReasonPromoted` given it's more generic language. See https://github.com/bluesky-social/social-app/pull/7518#issuecomment-2671965102

DGaffney commented:
Bumping on this thread - we're looking at adding subscriber tiers for feeds as an option on Graze, and using the `skeletonReason*` would be likely super helpful to denote who is a subscriber / sponsor in the feed, who helps pay for the feed etc
